pkgs/development/python-modules/opentelemetry-api/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, buildPythonPackage
+, pythonPackages
+, pythonOlder
+, fetchFromGitHub
+, deprecated
+, hatchling
+, importlib-metadata
+, opentelemetry-test-utils
+, setuptools
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "opentelemetry-api";
+  version = "1.18.0";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "open-telemetry";
+    repo = "opentelemetry-python";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-h6XDzM29wYiC51S7OpBXvWFCfZ7DmIyGMG2pFjJV7pI=";
+    sparseCheckout = [ "/${pname}" ];
+  } + "/${pname}";
+
+  format = "pyproject";
+
+  nativeBuildInputs = [
+    hatchling
+  ];
+
+  propagatedBuildInputs = [
+    deprecated
+    importlib-metadata
+    setuptools
+  ];
+
+  nativeCheckInputs = [
+    opentelemetry-test-utils
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "opentelemetry" ];
+
+  doCheck = false;
+
+  # Enable tests via passthru to avoid cyclic dependency with opentelemetry-test-utils.
+  passthru.tests.${pname} = pythonPackages.${pname}.overridePythonAttrs { doCheck = true; };
```

Comment:
This should either use finalAttrs or move the package in a let in and re-use that in passthru.

Reply:
I don't think `buildPythonPackage` supports the `finalAttrs` fixed-point approach like `mkDerivation` does. I tried that first. I'll switch to `let`.